fda/fdata_boxplot.py

```diff
@@ -0,0 +1,627 @@
+                if fig and ax are None.
+
+         Returns:
+            fig (figure object): figure object in which the graphs are plotted in case ax is None.
+            ax (axes object): axes in which the graphs are plotted.
+
+        """
+
+        fig, ax = self.fdatagrid.generic_plotting_checks(fig, ax, nrows,
+                                                         ncols)
+        tones = np.linspace(0.1, 1.0, len(self._prob) + 1, endpoint=False)[1:]
+        color = self.colormap(tones)
+
+        if self.fullout:
+            var_zorder = 4
+        else:
+            var_zorder = 1
+
+        for m in range(self.fdatagrid.ndim_image):
+
+            # outliers
+            for j in list(range(self.fdatagrid.nsamples)):
+                if self.outliers[m, j]:
+                    ax[m].plot(self.fdatagrid.sample_points[0],
+                               self.fdatagrid.data_matrix[j, :, m],
+                               color=self.outliercol,
+                               linestyle='--', zorder=1)
+
+            for i in range(len(self._prob)):
+                # central regions
+                ax[m].fill_between(self.fdatagrid.sample_points[0],
+                                   self.central_regions[
+                                       m * len(self._prob) + i, 0],
+                                   self.central_regions[
+                                       m * len(self._prob) + i, 1],
+                                   facecolor=color[i], zorder=var_zorder)
+
+            # outlying envelope
+            ax[m].plot(self.fdatagrid.sample_points[0],
+                       self.outlying_envelope[m, 0],
+                       self.fdatagrid.sample_points[0],
+                       self.outlying_envelope[m, 1], color=self.barcol,
+                       zorder=4)
+
+            # central envelope
+            ax[m].plot(self.fdatagrid.sample_points[0],
+                       self.central_envelope[m, 0],
+                       self.fdatagrid.sample_points[0],
+                       self.central_envelope[m, 1], color=self.barcol,
+                       zorder=4)
+
+            # vertical lines
+            index = math.ceil(self.fdatagrid.ncol / 2)
+            x = self.fdatagrid.sample_points[0][index]
+            ax[m].plot([x, x], [self.outlying_envelope[m, 0][index],
+                                self.central_envelope[m, 0][index]],
+                       color=self.barcol,
+                       zorder=4)
+            ax[m].plot([x, x], [self.outlying_envelope[m, 1][index],
+                                self.central_envelope[m, 1][index]],
+                       color=self.barcol, zorder=4)
+
+            # median sample
+            ax[m].plot(self.fdatagrid.sample_points[0], self.median[m],
+                       color=self.mediancol, zorder=5)
+
+        self.fdatagrid.set_labels(fig, ax)
+
+        return fig, ax
+
+    def _repr_svg_(self):
+        plt.ioff()
```

Comment:
I do not think this line is necessary.

```suggestion
```

Reply:
If this line is not included when the object is interpreted, two images appear: the figure of the plot method and its default representation via _repr_svg_. This line prevents the first one from appearing.

Reply:
No, because you do not plot in this method.

Reply:
``` def _repr_svg_(self):
        plt.ioff()
        fig, _ = self.plot()
        output = StringIO()
        fig.savefig(output, format='svg')
        plt.close(fig)
        return output.getvalue()``